src/SqlAsyncCollector.cs

```diff
@@ -403,8 +403,11 @@ private static void GenerateDataQueryForMerge(TableInformation table, IEnumerabl
 
             rowData = JsonConvert.SerializeObject(rowsToUpsert, table.JsonSerializerSettings);
             IEnumerable<string> columnNamesFromItem = GetColumnNamesFromItem(rows.First());
-            IEnumerable<string> bracketColumnDefinitionsFromItem = table.Columns.Where(c => columnNamesFromItem.Contains(c.Key))
-                .Select(c => $"{c.Key.AsBracketQuotedString()} {c.Value}");
+            var bracketColumnDefinitionsFromItem = new List<string>();
+            foreach (string columnName in columnNamesFromItem)
+            {
+                bracketColumnDefinitionsFromItem.Add($"{columnName.AsBracketQuotedString()} {table.Columns[columnName]}");
+            }
```

Comment:
(might want to do this one yourself, easy to make typos in Github...)

```suggestion
            IEnumerable<string> bracketColumnDefinitionsFromItem = columnNamesFromItem.Select(c => $"{c.AsBracketQuotedString()} {table.Columns[columnName]}");
            }
```